pkg/kubelet/apis/device-plugin/v1alpha1/api.proto

```diff
@@ -0,0 +1,106 @@
+
+option (gogoproto.goproto_stringer_all) = false;
+option (gogoproto.stringer_all) =  true;
+option (gogoproto.goproto_getters_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.sizer_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (gogoproto.goproto_unrecognized_all) = false;
+
+
+// Registration is the service advertised by the Kubelet
+// Only when Kubelet answers with a success code to a Register Request
+// may Device Plugins start their service
+service Registration {
+	rpc Register(RegisterRequest) returns (RegisterResponse) {}
+}
+
+// DevicePlugin is the service advertised by Device Plugins
+service DevicePlugin {
+	// ListWatch returns a stream of List of Devices
+	// Whenever a Device state change or a Device disapears, ListWatch
+	// returns the new list
+	rpc ListWatch(Empty) returns (stream ListWatchResponse) {}
+
+	// Allocate is called during container creation so that the Device
+	// Plugin can run device specific operations and instruct Kubelet
+	// of the steps to make the Device available in the container
+	rpc Allocate(AllocateRequest) returns (AllocateResponse) {}
+}
+
+message RegisterRequest {
+	// Version of the API the Device Plugin was built against
+	string version = 1;
+	// Name of the unix socket the device plugin is listening on
+	string endpoint = 2;
+	// Name of the Device advertised
+	string resourceName = 3;
+	// The labels to advertise at the node level
+	map<string, string> labels = 4;
+}
+
+message RegisterResponse {
+	// Minimum version the Kubelet API supports.
+	bool success = 1;
+	// Kubelet fills this field if it encounters any errors
+	// during the registration process
+	string error = 2;
+}
+
+// Allocate is called during container creation so that the Device
+// Plugin can run device specific operations on the Devices sent
+message AllocateRequest {
+	repeated string devicesIDs = 1;
+}
+
+message AllocateResponse {
+	repeated DeviceRuntimeSpec spec = 1;
+}
+
+// ListWatch returns a stream of List of Devices
+// Whenever a Device state change or a Device disapears, ListWatch
+// returns the new list
+message ListWatchResponse {
+	repeated Device devices = 1;
+}
+
+// The list to be added to the CRI spec
+message DeviceRuntimeSpec {
+	// List of environment variable to set in the container.
+	map<string, string> envs = 1;
+	// Mounts for the container.
+	repeated Mount mounts = 2;
+}
+
+// Mount specifies a host volume to mount into a container.
+// where device library or tools are installed on host and container
+message Mount {
+	// Path of the mount on the host.
+	string host_path = 1;
+	// Path of the mount within the container.
+	string mount_path = 2;
+	// If set, the mount is read-only.
+	bool read_only = 3;
+}
+
+/* E.g:
+* struct Device {
+*    ID: "GPU-fef8089b-4820-abfc-e83e-94318197576e",
+*    State: "Healthy",
+*}
+*/
+message Device {
+	string ID = 2;
+	string state = 3;
+
+	DeviceRuntimeSpec spec = 5;
```

Comment:
Thanks! To make allocate phase optional, we need extra information passed to Kubelet. Since we don't have that yet, probably just don't support optional Allocate for now.